include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_base.h

```diff
@@ -51,6 +69,12 @@ class PreconditionerBase
   {
     return std::make_shared<TimerTree>();
   }
+
+private:
+  virtual void
+  do_initialize() = 0;
```

Comment:
I guess this function will eventually be `protected` rather than `private`, because we expect the derived classes to override the function?